src/telegraf.ts

```diff
@@ -38,9 +38,39 @@ namespace Telegraf {
   }
 
   export interface LaunchOptions {
-    // FIXME: not honored by webhook
     /** List the types of updates you want your bot to receive */
     allowedUpdates?: tt.UpdateType[]
+    /**
+     * **In long polling mode,** when your bot receives one or more updates,
+     * `telegraf` calls the middleware stack and waits for it to complete before
+     * calling `getUpdates` again. This is important: calling `getUpdates`
+     * [confirms the previously received updates to Telegram via the `offset`
+     * parameter](https://core.telegram.org/bots/api#getupdates), but if your
+     * middleware crashes, you may want to request them again, which would
+     * become impossible then.
+     *
+     * The `concurrency` flag (default: `false`) allows you to adjust this
+     * behavior. It determines the maximum number of updates your bot may
+     * already be processing concurrently before making the next `getUpdates`
+     * request. The higher the number, the earlier `telegraf` will make the next
+     * request. **Caution:** setting this value to anything greater than `1` may
+     * cause your bot to miss updates if it crashes!
+     *
+     * **In webhook mode,** it is possible to receive old updates again while
+     * new ones have already been processed, so this is not an issue. The flag
+     * behaves exactly the same way, but its default value is `true`. If you set
+     * it to a lower value, `telegraf` will simply not answer the HTTP requests
+     * until enough of your middleware completed.
+     *
+     * `false` is an alias for `1`. `true` is an alias for `Infinity`. Values
+     * below `1` will be interpreted as `1`.
+     */
+    concurrency?: boolean | number
+    /** Configuration options for when the bot is run via long polling */
+    polling?: {
+      /** Determines the timout of the `getUpdates` calls */
+      timeout?: number
```

Comment:
Why is this needed?

Reply:
This determines the timeout of the `getUpdates` calls.

We [wanted](https://github.com/telegraf/telegraf/discussions/1234#discussioncomment-208597) to make all three timeouts configurable:
- `getUpdates` calls
- middleware
- fetching

Middleware timeouts can be configured by `handlerTimeout`. If I'm not mistaken, we do not support setting a timeout for fetching (yet).